Jenkinsfile

```
@@ -33,7 +33,7 @@ node('rhel8') {
         stage('Snapshot') {
             def filesToPush = findFiles(glob: '**.tgz')
             for (i = 0; i < filesToPush.length; i++) {
-                sh "rsync -Pzrlt --rsh=ssh --protocol=28 ${filesToPush[i].path} ${UPLOAD_LOCATION}/snapshots/vscode-wizard/"
+                sh "sftp ${UPLOAD_LOCATION}/snapshots/vscode-wizard/ <<< \$'put ${filesToPush[i].path}'"
```

uses -p (for preserving timestamp and file permissions), -r for recursive for put command and -C for compression for sftp command.
also -q for quiet could be nice.
warning, there is no equivalent for preserving or handle source symbolic links ( -l on rsync )

---

```suggestion
                sh "sftp -C ${UPLOAD_LOCATION}/snapshots/vscode-wizard/ <<< \$'put -p -r ${filesToPush[i].path}'"
```

something like that?

---

@sbouchet I am not sure if I would want that. 
Using `-r` flag could be usable when one it copying content of the folder, but as it was pointed out elsewhere, there must be a folder created first using `mkdir`. 
I would not use `-q` as it might be nice to be familiar about what is going on during transfer, at least in the beginning of using `sftp `.